app/com/linkedin/drelephant/schedulers/AirflowScheduler.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.schedulers;
+
+import com.linkedin.drelephant.configurations.scheduler.SchedulerConfigurationData;
+import com.linkedin.drelephant.util.Utils;
+
+import java.util.Properties;
+import org.apache.log4j.Logger;
+
+
+/**
+ * This class provides methods to load information specific to the Airflow scheduler.
+ */
+public class AirflowScheduler implements Scheduler {
+
+  private static final Logger logger = Logger.getLogger(AirflowScheduler.class);
+
+  public static final String AIRFLOW_TASK_ID = "airflow.ctx.task.task_id";
+  public static final String AIRFLOW_TASK_INSTANCE_EXECUTION_DATE = "airflow.ctx.task_instance.execution_date";
+  public static final String AIRFLOW_DAG_ID = "airflow.ctx.dag.dag_id";
+  public static final String AIRFLOW_DAG_RUN_EXECUTION_DATE = "airflow.ctx.dag_run.execution_date";
+
+  public static final String AIRFLOW_BASE_URL_PARAM_NAME = "airflowbaseurl";
+  private static final String AIRFLOW_BASE_URL_DEFAULT = "http://localhost:8000";
+
+  private String _schedulerName;
+  private String _taskId;
+  private String _taskInstanceExecutionDate;
+  private String _dagId;
+  private String _dagRunExecutionDate;
+  private int _subdagDepth;
+  private String _baseUrl;
+
+
+  public AirflowScheduler(String appId, Properties properties, SchedulerConfigurationData schedulerConfData) {
+    _schedulerName = schedulerConfData.getSchedulerName();
+    _baseUrl = schedulerConfData.getParamMap().getOrDefault(AIRFLOW_BASE_URL_PARAM_NAME, AIRFLOW_BASE_URL_DEFAULT);
```

Comment:
Realized that this isn't going to compile with older versions of Java. @akshayrai which versions do we want to support? It would be good if that were in the wiki somewhere.